client/src/api/datasource/queries.ts

```diff
@@ -58,6 +58,20 @@ export const useQueryMeta = (type: string, queryString: string, enabled = true)
   );
 };
 
+export const useQueryTrack = (type: string, account: string, container: string, filepath: string, enabled = true) => {
+  const { dataSourcesQuery, filesQuery } = useUserSettings();
+  return useQuery<Track>(
+    ['track', account, container, filepath],
+    async ({ signal }) => {
+      const client = DataSourceClientFactory(type, filesQuery.data, dataSourcesQuery.data);
+      return await client.track(account, container, filepath);
+    },
+    {
+      enabled: enabled,
+    }
+  );
+};
```

Comment:
Same thingg would make more sense to have this inside metadata and use the signal as part of the parameters you pass to track.

Reply:
aye, done